### PR TITLE
Fix the handling of the file browser configuration

### PR DESCRIPTION
**Pull request by jtpio**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/14623

## Code changes

Move the handling of the file browser settings to a separate plugin, so it's easier to change the settings and not have them collide in downstream applications such as Notebook 7.

## User-facing changes

None for JupyterLab users.

For Notebook users, using https://github.com/jupyter/notebook/pull/7481, the file browser will show different defaults than JupyterLab (checkboxes, file size column, single click navigation), without having to modify the settings and have these defaults be reflected in JupyterLab too:

[filebrowser-settings-notebook-lab.webm](https://github.com/user-attachments/assets/001df8eb-7c08-404d-bbee-22d414b0e34d)

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jtpio** [comment]
Adding to `4.3.0` as this is actually a bug fix (and hoping to finish the PR before the final release).

**Comment by jtpio** [comment]
hmm the UI test failures seem to be consistent but are not expected, need to check what is happening there:

example-diff

**Comment by jtpio** [comment]
On a local install from `main`, loading JupyterLab would set the following size for the file browser columns:

image

Switching to this branch:

image

So moving the logic to a separate plugin seems to introduce a small delay that has some effect on the final columns widths, which also seems to be reflected in the UI tests.

Maybe it's then still fine to update the reference snapshots, as the differences are quite minor in the end.

**Comment by jtpio** [comment]
bot please update snapshots

**Comment by jtpio** [comment]
bot please update snapshots

**Comment by jtpio** [comment]
Looks like CI is now passing :heavy_check_mark: 

If we are fine with the minor updates to the snapshots, this should be ready to go.

File: packages/filebrowser-extension/src/index.ts [hunk 1]
```diff
@@ -160,7 +160,7 @@ const namespace = 'filebrowser';
  */
 const browser: JupyterFrontEndPlugin<IFileBrowserCommands> = {
   id: FILE_BROWSER_PLUGIN_ID,
-  description: 'Set up the default file browser (commands, settings,...).',
+  description: 'Set up the default file browser commands',
```
Comment (krassowski) [inline packages/filebrowser-extension/src/index.ts:163]:
```suggestion
  description: 'Set up the default file browser commands and state restoration',
```

Comment (krassowski) [reply packages/filebrowser-extension/src/index.ts:163]: Since it also handles the `restorer` and `preferredPath`.

**Pull request merged**